examples/Fine_tuning_for_function_calling.ipynb

```diff
@@ -0,0 +1,1248 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Fine tuning for function calling (with synthetic data)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This notebook covers how to finetune for function calling to increase function calling accuracy and reliability.\\\n",
+    "For reference, the cookbook recipe for function calling is here: https://github.com/openai/openai-cookbook/blob/main/examples/How_to_call_functions_with_chat_models.ipynb \\\n",
+    "and a cookbook recipe for fine tuning chat responses is here: https://github.com/openai/openai-cookbook/blob/main/examples/How_to_finetune_chat_models.ipynb"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "For context, from the function calling notebook above:\n",
+    "> `functions` is an optional parameter in the Chat Completion API which can be used to provide function specifications. The purpose of this is to enable models to generate function arguments which adhere to the provided specifications. Note that the API will not actually execute any function calls. It is up to developers to execute function calls using model outputs."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Overview\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This notebook contains three sections\n",
```

Comment:
Before we even get here it could be good to motivate the entire notebook – _why_ should someone do fine-tuning with function calling? Why are we doing these steps? (maybe even how we got here?)